Cython/Compiler/Symtab.py

```diff
@@ -2012,6 +2034,28 @@ def __init__(self, name, scope_name, outer_scope, parent_scope=None):
     def declare_pyfunction(self, name, pos, allow_redefine=False):
         return LocalScope.declare_pyfunction(self, name, pos, allow_redefine, visibility='private')
 
+    def declare_assignment_expression_target(self, name, type, pos):
+        if self.is_generator_expression_scope:
+            entry = self.parent_scope.declare_var(name, type, pos)
+            entry.in_closure = True
+            inner_entry = InnerEntry(entry, self)
+            inner_entry.is_variable = True
+            self.entries[name] = inner_entry
+            return inner_entry
+        return self.declare_var(name, type, pos)
+
+    def lookup_assignment_expression_target(self, name):
+        entry = self.lookup_here(name)
+        if not entry and self.is_generator_expression_scope:
+            entry = self.parent_scope.lookup_assignment_expression_target(name)
+            if entry:
+                entry.in_closure = True
+                inner_entry = InnerEntry(entry, self)
+                inner_entry.is_variable = True
+                self.entries[name] = inner_entry
+                return inner_entry
+        return entry
```

Comment:
This suggests subclassing to me.

Reply:
You mean `is_generator_expression_scope` should be linked to a subclass? Or something else?

Reply:
Yes. The way you use the flag pretty much cuts these methods into two entirely separate implementations. That is much better represented with a subtype. And a `GeneratorExpressionScope` seems a reasonable type to have.

Reply:
Done